File: LeetCode/27. Remove Element/27.java
class Solution {
    public int removeElement(int[] nums, int val) {
        // Initialize the counter k
        int k = 0;

        // Iterate over the array
        for(int num : nums){
            // If the current element is not equal to the value to be removed
            if(num != val){
                // Replace the element at the position of the counter with the current element
                nums[k] = num;
                // Increment the counter
                k++;
            }
        }

        // Return the counter
        return k;
    }
}

File: LeetCode/27. Remove Element/27.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        # Initialize the counter k
        k = 0

        # Iterate over the list
        for num in nums:
            # If the current element is not equal to the value to be removed
            if num != val:
                # Replace the element at the position of the counter with the current element
                nums[k] = num
                # Increment the counter
                k += 1

        # Return the counter
        return k